api/handlers.go

```diff
@@ -487,6 +487,15 @@ func (si *ServerImplementation) SearchForTransactions(ctx echo.Context, params g
 		return badRequest(ctx, err.Error())
 	}
 
+	if filter.AddressRole != 0 {
+		var address basics.Address
+		copy(address[:], filter.Address)
+		if address.IsZero() {
+			return badRequest(
+				ctx, "searching transactions by zero address with address role is not supported")
+		}
+	}
+
```

Comment:
This should go into `transactionParamsToTransactionFilter` or `si.fetchTransactions` to make sure the constraint is checked for all account searches.

Reply:
Only `LookupApplicationLogsByID()` would violate, which I think is fine. If we do put it in `fetchTransactions()`, `LookupApplicationLogsByID()` would return a weird error.

Reply:
`transactionParamsToTransactionFilter()` is a conversion function, I don't want to put business logic there.

I think the current code is ok, especially given that no one in their right mind will use `LookupApplicationLogsByID()` with zero address.

Reply:
It also validates the input, there's a test in `api/handlers_test.go` for it also so you can add a test:
```
    {
      name: "Invalid zero address parameters",
      params: generated.SearchForTransactionsParams{
        Address:     strPtr(basics.Address{}.String()),
        AddressRole: 9,
      },
      filter:        idb.TransactionFilter{},
      errorContains: []string{errInvalidZeroAddressSearch},
    }
```

Could you also add the error message to `error_messages.go`?

Reply:
Moved the error message to `error_messages.go`. `transactionParamsToTransactionFilter()` does a couple of sanity checks, but no more. Definitely no artificial limitations should be in there.

Reply:
I think it is incorrect that https://github.com/algorand/indexer/blob/830e1ba83500b2c78251070fa67031efdd418ed7/api/handlers.go#L277 becomes `idb.AddressRoleSender | idb.AddressRoleAssetSender`, so I set `filter.AddressRole` directly and run validation that used to be in `transactionParamsToTransactionFilter()` after.